packages/web/src/pages/sign-up-page/components/HandleField.tsx

```diff
@@ -39,18 +48,53 @@ type HandleFieldProps = Partial<HarmonyTextFieldProps> & {
   onErrorSocialMediaLogin?: () => void
 }
 
+/**
+ * This is a workaround because Formik's isValidating only returns true DURING the validation.
+ * However for this scenario we want to wait for the results of the validation before showing any messaging.
+ * To do this we wait for Formik isValidating to change to true, then wait for it to change
+ * back to false before saying we're "not waiting" for validation anymore
+ * @returns {isWaitingOnValidation, setIsWaitingOnValidation}
+ */
+const useIsWaitingOnValidation = () => {
+  const { isValidating } = useFormikContext()
+  const [isWaitingOnValidation, setIsWaitingOnValidation] = useState(false)
+  const [previousIsValidating, setPreviousIsValidating] = useState(false)
+
+  // Wait for formik to start validating
+  useEffect(() => {
+    if (isValidating && isWaitingOnValidation) {
+      setPreviousIsValidating(true)
+    }
+  }, [isValidating, isWaitingOnValidation])
+
+  // We know formik has stopped validating when it differs from our previous value
+  useEffect(() => {
+    if (previousIsValidating && !isValidating) {
+      setIsWaitingOnValidation(false)
+      setPreviousIsValidating(false) // reset the value
+    }
+  }, [isValidating, previousIsValidating])
+
+  return { isWaitingOnValidation, setIsWaitingOnValidation }
+}
+
```

Comment:
this feels "wrong" but I dont see an "out of the box" way with the stuff Formik provides 🤔

Reply:
For sure I had same issue with email taken validation, check out how it's done there cause I think it's nearly the same use case. Might be worth consolidation

Reply:
Looked into it, they're similar but not quite the same. Definitely helped to use `usePrevious` tho